src/apps/MASS3DPA.cpp

```diff
@@ -0,0 +1,95 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "MASS3DPA.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#include "common/DataUtils.hpp"
+
+namespace rajaperf
+{
+namespace apps
+{
+
+
+MASS3DPA::MASS3DPA(const RunParams& params)
+  : KernelBase(rajaperf::Apps_MASS3DPA, params)
+{
+  m_NE_default = 924385;
+
+  setDefaultSize(m_NE_default);
+  setDefaultReps(50);
+
+  setProblemSize( getRunSize() * Q1D * Q1D);
+
+  m_NE = (getProblemSize())/(Q1D*Q1D);
```

Comment:
guard against getting 0, when problem size is less than Q1D*Q1D

Reply:
move to line 28.